xbmc/pvr/epg/EpgSearchFilter.cpp

```diff
@@ -346,6 +361,26 @@ bool CPVREpgSearchFilter::MatchChannel(const std::shared_ptr<CPVREpgInfoTag>& ta
          CServiceBroker::GetPVRManager().Clients()->IsCreatedClient(tag->ClientID());
 }
 
+bool CPVREpgSearchFilter::MatchChannelGroup(const std::shared_ptr<CPVREpgInfoTag>& tag) const
+{
+  if (m_iChannelGroupID != -1)
+  {
+    if (!m_groupIdMatches.has_value())
+    {
+      const std::shared_ptr<CPVRChannelGroup> group = CServiceBroker::GetPVRManager()
+                                                          .ChannelGroups()
+                                                          ->Get(m_bIsRadio)
+                                                          ->GetById(m_iChannelGroupID);
```

Comment:
I know clang format gives a lot of benefits but this is super ugly!

Reply:
Maybe I can trick clang-format by using `auto` ?

Reply:
No luck. So we will have to live this, even though I agree it's ugly.